mkosi/sandbox.py

```diff
@@ -438,6 +443,67 @@ def is_relative_to(one: str, two: str) -> bool:
     return os.path.commonpath((one, two)) == two
 
 
+def pack_file_descriptors() -> int:
+    fds = []
+
+    with os.scandir("/proc/self/fd") as it:
+        for e in sorted(it, key=lambda e: e.name):
+            if not e.is_symlink() and (e.is_file() or e.is_dir()):
+                continue
+
+            try:
+                fd = int(e.name)
+            except ValueError:
+                continue
+
+            if fd < SD_LISTEN_FDS_START:
+                continue
+
+            fds.append(fd)
+
+    # os.scandir() either opens a file descriptor to the given path or dups the given file descriptor. Either
+    # way, there will be an extra file descriptor in the fds array that's not valid anymore now, so find out
+    # which one and drop it.
+    fds = [fd for fd in fds if libc.fcntl(fd, F_GETFD, 0) >= 0]
```

Comment:
Doing the `sorted` call here instead would sidestep the need for a key function.
```suggestion
    fds = sorted(fd for fd in fds if libc.fcntl(fd, F_GETFD, 0) >= 0)
```